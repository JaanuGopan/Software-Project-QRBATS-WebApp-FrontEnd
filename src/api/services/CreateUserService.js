import axios from "axios";
import ApiConstants from "../config/ApiConstants";

class CreateUserService {
  static async saveUser(
    firstName,
    lastName,
    email,
    password,
    userName,
    departmentId,
    role
  ) {
    const deparmentList = ["DEIE", "DCOM", "DMME", "DCEE", "DMENA"];
    const userRoleList = ["ADMIN", "LECTURER"];
    try {
      const response = await axios.post(ApiConstants.signupUrl, {
        firstName: firstName,
        lastName: lastName,
        email: email,
        password: password,
        userName: userName,
        departmentId: departmentId,
        role: role,
      });
      return response.data;
    } catch (error) {
      throw new Error("The User is not created. ", error);
    }
  }
}

export default CreateUserService;
