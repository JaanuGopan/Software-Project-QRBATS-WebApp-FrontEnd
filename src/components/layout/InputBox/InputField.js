import React from "react";
import "./inputfield.css";

const InputField = ({ placeholder, value, onChange, inputType }) => {
  return (
    <div className="input-with-icon">
      <input
        required
        type={inputType}
        className="input-field"
        placeholder={placeholder}
        value={value}
        onChange={onChange}
      />
    </div>
  );
};

export default InputField;
