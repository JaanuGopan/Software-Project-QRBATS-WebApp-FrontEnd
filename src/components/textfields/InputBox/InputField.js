import React from "react";
import "../../../pages/Event/EventCreation/EventCreation.css";

const InputField = ({ placeholder, value, onChange, inputType }) => {
  return (
    <div className="input-with-icon">
      <input
        required
        type={inputType}
        className="form-control mb-2"
        placeholder={placeholder}
        value={value}
        onChange={onChange}
      />
    </div>
  );
};

export default InputField;
