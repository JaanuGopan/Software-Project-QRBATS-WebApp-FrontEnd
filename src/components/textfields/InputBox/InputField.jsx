import React from 'react';
import '../../../pages/Event/EventCreation/EventCreation.css';

const InputField = ({ placeholder, value, onChange, inputType, error, helperText }) => {
  return (
    <div className="input-with-icon">
      <input
        required
        type={inputType}
        className={`form-control mb-2 ${error ? 'is-invalid' : ''}`}
        placeholder={placeholder}
        value={value}
        onChange={onChange}
      />
      {error && <div className="invalid-feedback">{helperText}</div>}
    </div>
  );
};

export default InputField;
