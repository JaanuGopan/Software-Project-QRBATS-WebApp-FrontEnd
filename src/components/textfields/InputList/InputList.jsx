import React from 'react';
import '../../../pages/Event/EventCreation/EventCreation.css';

const InputList = ({ placeholder, value, onChange, inputType, list, initialValue }) => {
  return (
    <div className="input-with-icon">
      <select
        required
        value={value}
        onChange={onChange}
        placeholder={placeholder}
        className="form-control mb-2"
      >
        {initialValue ? null : <option value="">{initialValue}</option>}
        {[initialValue, ...list.filter((option) => option !== initialValue)].map(
          (option, index) => (
            <option key={index} value={option} selected={option === initialValue}>
              {option}
            </option>
          )
        )}
      </select>
    </div>
  );
};

export default InputList;
