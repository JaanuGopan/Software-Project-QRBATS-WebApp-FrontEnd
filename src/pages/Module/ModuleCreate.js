import React, { useState } from "react";
import "./ModulePage.css";
import Designer from "../../assets/Images/Designer.jpeg";
import InputField from "../../components/textfields/InputBox/InputField";
import { IoMdCloseCircleOutline } from "react-icons/io";
import CreateUserService from "../../api/services/CreateUserService";
import InputList from "../../components/textfields/InputList/InputList";

const ModuleCreate = ({ handleCloseModuleCreateWindow }) => {
  const [moduleName, setModuleName] = useState("");
  const [moduleCode, setModuleCode] = useState("");
  const [semester, setSemester] = useState("");
  const [moduleEnrolmentKey, setModuleEnrolmentKey] = useState("");
  const [departmentId, setDepartmentId] = useState("");
  const [lectureId, setLectureId] = useState("");

  const departmentList = ["DEIE", "DCOM", "DMME", "DCEE", "DMENA"];
  const semesterList = ["1", "2", "3", "4", "5", "6", "7", "8"];

  const handleSubmit = async (e) => {
    e.preventDefault();
    try {
      const response = CreateUserService.saveUser();

      console.error("Login success...");
      handleCloseModuleCreateWindow();
    } catch (error) {
      console.error("Login failed", error);
    }
  };

  return (
    <div className="module-signup-main-container">
      <div
        className="closeCreateEventWindow"
        onClick={handleCloseModuleCreateWindow}
      >
        <IoMdCloseCircleOutline />
      </div>
      <p className="module-head1">Create module</p>
      <div className="module-login-container">
        <div className="module-image-container">
          <img src={Designer} className="module-logo" alt="Logo" />
        </div>
        <div className="form-container">
          <form onSubmit={handleSubmit}>
            {/* First Name Input */}
            <InputField
              placeholder="Enter The Module Name"
              value={moduleName}
              onChange={(e) => setModuleName(e.target.value)}
              inputType="text"
            />

            {/* Last Name Input */}
            <InputField
              placeholder="Enter the module code"
              value={moduleCode}
              onChange={(e) => setModuleCode(e.target.value)}
              inputType="text"
            />

            <div className="choice-input mb-3">
              <InputList
                placeholder="Select semester"
                value={semester}
                onChange={(e) => setSemester(e.target.value)}
                list={semesterList}
                inputType="text"
                initialValue="Select semester"
                className="student-select-input"
              />
            </div>

            <div className="choice-input mb-3">
              <InputList
                placeholder="Select department"
                value={departmentId}
                onChange={(e) => setDepartmentId(e.target.value)}
                list={departmentList}
                inputType="text"
                initialValue="Select department"
                className="student-select-input"
              />
            </div>

            {/* Password Input */}
            <InputField
              placeholder="Enter Module Enrolment Key"
              value={moduleEnrolmentKey}
              onChange={(e) => setModuleEnrolmentKey(e.target.value)}
              inputType="password"
            />

            <button type="submit" className="btn btn-success w-100">
              Create module
            </button>
          </form>
        </div>
      </div>
    </div>
  );
};

export default ModuleCreate;
