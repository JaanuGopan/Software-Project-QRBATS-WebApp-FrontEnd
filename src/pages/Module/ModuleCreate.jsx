import React, { useState } from 'react';
import './ModulePage.css';
import Designer from '../../assets/Images/Designer.jpeg';
import InputField from '../../components/textfields/InputBox/InputField';
import { IoMdCloseCircleOutline } from 'react-icons/io';
import { IoIosClose } from 'react-icons/io';
import InputList from '../../components/textfields/InputList/InputList';
import ModuleService from '../../api/services/ModuleService';
import { selectUser } from '../../redux/features/userSlice';
import { useSelector } from 'react-redux';
import { ToastContainer, toast } from 'react-toastify';
import Select from 'react-select';
import { CircularProgress } from '@mui/material';

const ModuleCreate = ({
  handleCloseModuleCreateWindow,
  handleReloadModuleList,
  handleModuleCreate,
}) => {
  const [moduleName, setModuleName] = useState('');
  const [moduleCode, setModuleCode] = useState('');
  const [semester, setSemester] = useState('');
  const [moduleEnrolmentKey, setModuleEnrolmentKey] = useState('');
  const [departmentId, setDepartmentId] = useState('');

  const { userId } = useSelector(selectUser) || {};

  const departmentList = ['DEIE', 'DCOM', 'DMME', 'DCEE', 'DMENA', 'DIS'];
  const semesterList = ['1', '2', '3', '4', '5', '6', '7', '8'];

  const [processing, setProcessing] = useState(false);

  const handleCreateModule = async (e) => {
    e.preventDefault();
    const moduleData = {
      moduleCode: moduleCode,
      moduleName: moduleName,
      semester: semester,
      moduleEnrolmentKey: moduleEnrolmentKey,
      departmentId: departmentList.indexOf(departmentId) + 1,
      userId,
    };
    try {
      setProcessing(true);
      await handleModuleCreate(moduleData);
    } finally {
      setProcessing(false);
    }
  };

  const handleClearData = () => {
    setModuleCode('');
    setModuleName('');
    setDepartmentId('');
    setModuleEnrolmentKey('');
    setSemester('');
  };

  return (
    <div className="module-create-main-container">
      <div className="module-create-title-close-button">
        <h3 className="module-create-title">Create Module</h3>
        <div className="module-create-close-button" onClick={handleCloseModuleCreateWindow}>
          <IoMdCloseCircleOutline id="close-icon" />
        </div>
      </div>
      <div className="module-login-container">
        <div className="form-container">
          <form onSubmit={handleCreateModule}>
            <div className="module-creation-input">
              <label>Module Name</label>
              <div className="module-creation-input-field">
                <InputField
                  placeholder="Enter The Module Name"
                  value={moduleName}
                  onChange={(e) => setModuleName(e.target.value)}
                  inputType="text"
                />
              </div>
            </div>

            <div className="module-creation-input">
              <label>Module Code</label>
              <div className="module-creation-input-field">
                <InputField
                  placeholder="Enter the module code"
                  value={moduleCode}
                  onChange={(e) => setModuleCode(e.target.value)}
                  inputType="text"
                />
              </div>
            </div>

            <div className="module-creation-input">
              <label>Semester</label>
              <div className="module-creation-input-field">
                <div className="choice-input mb-3">
                  <InputList
                    placeholder="Select semester"
                    value={semester}
                    onChange={(e) => setSemester(e.target.value)}
                    list={semesterList}
                    inputType="text"
                    initialValue="Select semester"
                    className="student-select-input"
                  />
                </div>
              </div>
            </div>

            <div className="module-creation-input">
              <label>Department</label>
              <div className="module-creation-input-field">
                <div className="choice-input mb-3">
                  <InputList
                    placeholder="Select department"
                    value={departmentId}
                    onChange={(e) => setDepartmentId(e.target.value)}
                    list={departmentList}
                    inputType="text"
                    initialValue="Select department"
                    className="student-select-input"
                  />
                </div>
              </div>
            </div>

            <div className="module-creation-input">
              <label>Enrolment Key</label>
              <div className="module-creation-input-field">
                <InputField
                  placeholder="Enter Module Enrolment Key"
                  value={moduleEnrolmentKey}
                  onChange={(e) => setModuleEnrolmentKey(e.target.value)}
                  inputType="text"
                />
              </div>
            </div>
            <div className="create-module-create-button">
              {processing ? (
                <CircularProgress />
              ) : (
                <button type="submit" className="btn btn-success">
                  Create module
                </button>
              )}
            </div>
          </form>
        </div>
      </div>
    </div>
  );
};

export default ModuleCreate;
