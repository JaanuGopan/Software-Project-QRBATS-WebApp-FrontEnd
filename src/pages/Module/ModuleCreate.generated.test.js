import renderer from 'react-test-renderer';
import React, { useState } from "react";

import Designer from "../../assets/Images/Designer.jpeg";
import InputField from "../../components/textfields/InputBox/InputField";
import { IoMdCloseCircleOutline } from "react-icons/io";
import CreateUserService from "../../api/services/CreateUserService";
import InputList from "../../components/textfields/InputList/InputList";
import ModuleCreate from "./ModuleCreate";

jest.mock("./ModulePage.css");
jest.mock("../../assets/Images/Designer.jpeg");
jest.mock("../../components/textfields/InputBox/InputField");
jest.mock("react-icons/io");
jest.mock("../../api/services/CreateUserService");
jest.mock("../../components/textfields/InputList/InputList");

const renderTree = tree => renderer.create(tree);
describe('<ModuleCreate>', () => {
  it('should render component', () => {
    expect(renderTree(<ModuleCreate 
    />).toJSON()).toMatchSnapshot();
  });
  
});