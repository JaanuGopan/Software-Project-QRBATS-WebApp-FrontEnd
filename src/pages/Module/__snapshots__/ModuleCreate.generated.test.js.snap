// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ModuleCreate> should render component 1`] = `
<div
  className="module-signup-main-container"
>
  <div
    className="closeCreateEventWindow"
  />
  <p
    className="module-head1"
  >
    Create module
  </p>
  <div
    className="module-login-container"
  >
    <div
      className="module-image-container"
    >
      <img
        alt="Logo"
        className="module-logo"
        src="Designer.jpeg"
      />
    </div>
    <div
      className="form-container"
    >
      <form
        onSubmit={[Function]}
      >
        <div
          className="choice-input mb-3"
        />
        <div
          className="choice-input mb-3"
        />
        <button
          className="btn btn-success w-100"
          type="submit"
        >
          Create module
        </button>
      </form>
    </div>
  </div>
</div>
`;
